src/client.rs

```diff
@@ -104,10 +110,20 @@ pub fn connect_with_config<Req: IntoClientRequest>(
     let addrs = (host, port).to_socket_addrs()?;
     let mut stream = connect_to_some(addrs.as_slice(), &request.uri(), mode)?;
     NoDelay::set_nodelay(&mut stream, true)?;
-    client_with_config(request, stream, config).map_err(|e| match e {
+    match client_with_config(request, stream, config).map_err(|e| match e {
         HandshakeError::Failure(f) => f,
         HandshakeError::Interrupted(_) => panic!("Bug: blocking handshake not blocked"),
-    })
+    }) {
+        Ok(r) => Ok(r),
+        Err(e) => match e {
+            Error::Redirection(uri) =>  {
+                debug!("Redirecting to {}", uri);
+                let request = request2.uri(uri).body(()).unwrap();
+                connect_with_config(request, config)
```

Comment:
We need to find a way to limit the amount of redirections to not end up with an infinite loop. Perhaps an additional parameter in the config would do? (if any)

Reply:
Sure thing, I'll add a limit in the next commit, what should the sensible default for it be?

Reply:
Maybe something like `max_redirect_count: Option<usize>`, where `None` would mean "not follow redirects", but I'm open to ideas :)